src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -146,8 +146,14 @@ export class PageControllerBase {
         if (isPageHeading) {
           labelOrLegend.isPageHeading = isPageHeading
 
+          // Check for optional in label
+          const isOptional =
+            this.components.formItems.at(0)?.options.required === false
+
           if (pageTitle) {
-            labelOrLegend.text = pageTitle
+            labelOrLegend.text = isOptional
+              ? pageTitle + ' (optional)'
```

Comment:
Mind using the export from **src/server/plugins/engine/components/constants.ts**?

```suggestion
              ? `${pageTitle}${optionalText}`
```

Reply:
done